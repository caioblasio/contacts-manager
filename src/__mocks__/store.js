const contactsState = require("providers/Contacts/reducer").initialState;

const history = require("history").createBrowserHistory();
const configureStore = require("store").default;

const rootState = {
  contacts: contactsState
};

const store = configureStore(rootState);

module.exports = store;
