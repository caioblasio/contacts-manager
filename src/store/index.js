import { createStore, applyMiddleware, compose } from "redux";
import { createBrowserHistory } from "history";
import { routerMiddleware } from "connected-react-router/immutable";
import createSagaMiddleware from "redux-saga";
import createRootReducer from "./reducers";
import rootSaga from "./sagas";

export const history = createBrowserHistory();

export default function configureStore(initialState = {}) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};
  if (ENVIRONMENT !== "production" && typeof window === "object") {
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        name: "Redux",
        realtime: true,
        trace: true,
        traceLimit: 25
      });
    }
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const middlewares = [sagaMiddleware, routerMiddleware(history)];
  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createRootReducer(history),
    initialState,
    composeEnhancers(...enhancers)
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
