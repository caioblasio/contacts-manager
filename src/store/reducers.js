import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router/immutable";
import contactsReducer from "providers/Contacts/reducer";

const createRootReducer = history =>
  combineReducers({
    contacts: contactsReducer,
    router: connectRouter(history)
  });
export default createRootReducer;
