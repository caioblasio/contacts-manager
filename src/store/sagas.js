import { all } from "redux-saga/effects";
import contactsSaga from "providers/Contacts/saga";

export default function*() {
  yield all([contactsSaga()]);
}
