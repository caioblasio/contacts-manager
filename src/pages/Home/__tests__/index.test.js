import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import HomeComponent from "pages/Home";

import store from "__mocks__/store";

describe("Given <HomeComponent />", () => {
  let component;
  let Page;

  beforeAll(() => {
    Page = props => (
      <Provider store={store}>
        <MemoryRouter>
          <HomeComponent {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  describe("When it initializes", () => {
    beforeEach(() => {
      const { container } = render(<Page />);
      component = container;
    });

    it("Then should match the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
