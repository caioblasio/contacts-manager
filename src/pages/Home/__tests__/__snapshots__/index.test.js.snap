// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <HomeComponent /> When it initializes Then should match the snapshot 1`] = `
<div>
  <section
    class="makeStyles-root-1"
  >
    <h1
      class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    >
      All Contacts
    </h1>
    <section />
    <a
      href="/new"
      style="text-decoration: none;"
    >
      <button
        aria-label="add"
        class="MuiButtonBase-root MuiFab-root makeStyles-floatButton-3 MuiFab-primary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </a>
  </section>
</div>
`;
